pkg/retrieval/retrieval.go

```diff
@@ -248,8 +202,24 @@ func (s *Service) RetrieveChunk(ctx context.Context, addr swarm.Address, origin
 	return v.(swarm.Chunk), nil
 }
 
-func (s *Service) retrieveChunk(ctx context.Context, addr swarm.Address, sp *skipPeers, originated bool) (chunk swarm.Chunk, peer swarm.Address, requested bool, err error) {
-	startTimer := time.Now()
+func (s *Service) retrieveChunk(ctx context.Context, resultChan chan retrievalResult, doneChan chan struct{}, addr swarm.Address, sp *skipPeers, originated bool) {
+
+	var (
+		peer       swarm.Address
+		err        error
+		chunk      swarm.Chunk
+		retrieved  bool
+		startTimer = time.Now()
+	)
+
+	defer func() {
```

Comment:
i wonder if this is needed here, maybe it is more readable if we inline this in the same place you're spawning the goroutine? then just keep this method the same with the normal return values?